File: repeated-string/repeated-string.py
def aCounter(s, l):
    i = a = 0
    
    while i < l:
        if s[i] == "a":
            a += 1
        i += 1
    return a

def repeatedString(s, n):
    l = len(s) 
    a = aCounter(s, l)
    result = (n / l) * a
    result += aCounter(s, n % l)
    return int(result)

def main():
    s = "a"
    n = 1000000000000
    print(repeatedString(s, n))

if __name__ == "__main__":
    main()

File: Warm-Up/repeated-string/repeated-string.c
#include <stdio.h>
#include <string.h>

int aCounter(char *s, int len)
{
	int a;
	
	a = 0;    
	while(len--)
	{
		if (*s == 'a')
			++a;
		++s;    
	}
	return (a);
}

long repeatedString(char* s, long n)
{
	int len;
	int a;
	long result;

	len = strlen(s);
	a = aCounter(s, len);
	result = (n / len) * a;
	result += aCounter(s, n % len);
	return (result);
}

int main(void)
{
	char	*s;
	long	n;

	s = "a";
	n = 1000000000000;
	printf("%ld\n", repeatedString(s, n));
	return (0);
}
